parsers/mysql/mysql.go

```diff
@@ -81,16 +97,81 @@ type SlowQuery struct {
 	LockTime        float64   `json:"lock_time"`
 	RowsSent        int       `json:"rows_sent"`
 	RowsExamined    int       `json:"rows_examined"`
-	Query           string    `json:"query",omitempty`
+	Query           string    `json:"query,omitempty"`
 	NormalizedQuery string    `json:"normalized_query,omitempty"`
+	DB              string    `json:"db,omitempty"`
+	HostedOn        string    `json:"hosted_on,omitempty"`
+	ReadOnly        bool      `json:"read_only,omitempty"`
 	skipQuery       bool
 }
 
-func (p *Parser) Init(_ interface{}) error {
+func (p *Parser) Init(options interface{}) error {
+	p.conf = *options.(*Options)
 	p.nower = &RealNower{}
+	if p.conf.Host != "" {
+		url := fmt.Sprintf("%s:%s@tcp(%s)/", p.conf.User, p.conf.Pass, p.conf.Host)
+		db, err := sql.Open("mysql", url)
+		if err != nil {
+			log.Fatal(err)
+		}
+		// update hostedOn and readOnly every <n> seconds
+		go func() {
+			defer db.Close()
+			ticker := time.NewTicker(time.Second * time.Duration(p.conf.QueryInterval))
+			for _ = range ticker.C {
+				p.hostedOn = getHostedOn(db)
+				p.readOnly = getReadOnly(db)
+			}
+		}()
+	}
 	return nil
 }
 
+func getReadOnly(db *sql.DB) bool {
+	rows, err := db.Query("SELECT @@global.read_only;")
+	if err != nil {
+		log.Fatal(err)
```

Comment:
log fatal is wrong from my side! good call! honeytail should not exit but
rather print the error!

On Wed, Aug 31, 2016 at 10:50 PM Ben Hartshorne <notifications@github.com>
wrote:

> In parsers/mysql/mysql.go
> <https://github.com/honeycombio/honeytail/pull/5#discussion_r77070545>:
>
> >  	return nil
> >  }
> >
> > +func getReadOnly(db *sql.DB) bool {
> > +	rows, err := db.Query("SELECT @@global.read_only;")
> > +	if err != nil {
> > +		log.Fatal(err)
>
> I'm a little concerned about the log.Fatal()s here. Our primary goal is
> parsing the log file and augmenting it with some extra data. If we have
> some transient error talking to the DB, is it reasonable to exit honeytail?
> Should we only exit if we fail several times in a row? or just leave read
> only and/or hosted on empty until they succeed the next time?
> I don't have a good understanding of the circumstances that might cause
> one of these calls (db.Query, rows.Scan, or rows.Err) to fail, so I'm not
> sure of the severity of this issue.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/honeycombio/honeytail/pull/5/files/cee72c3fdc98165d6ab4a43d821e7822ee1322f6#r77070545>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAHMTZisbbAHoUkDfCU8kZqOobBYE7tYks5qlekHgaJpZM4Jwzdc>
> .
>